### DISCUSSION TITLE
How to get the feature of 'End Task'?

### DISCUSSION THREAD

**Discussion by achyutghosh** [dahall/TaskScheduler#943] [Q&A] [answered]
I can disable the task scheduler using this  `task.Definition.Settings.Enabled = false;`. I want to end the task and then disable the scheduler. But I'm not sure how to do that, so can anyone share any code snippet for that or any doc links?

**Comment by dahall** [comment, marked as answer]
`task.Enabled = false` will disable the single task immediately. `task.Definition.Settings.Enabled = false` is used to disable a task before registration.

To terminate a running task, use:
```cs
if (task.IsActive && task.State == TaskState.Running)
   task.Stop();
```

**Reply by achyutghosh** [reply]
I've found this after a little research, but thank you for your generation reply.
